cli/args/lockfile.rs

```diff
@@ -114,25 +115,45 @@ pub async fn snapshot_from_lockfile(
   };
 
   // now that the lockfile is dropped, fetch the package version information
-  let mut version_infos =
-      FuturesOrdered::from_iter(packages.iter().map(|p| p.pkg_id.nv.clone()).map(
-        |nv| async move {
-          let package_info = api.package_info(&nv.name).await?;
-          match package_info.version_info(&nv) {
-            Ok(version_info) => Ok(version_info),
-            Err(err) => {
-              bail!("Could not find '{}' specified in the lockfile. Maybe try again with --reload", err.0);
-            }
-          }
-        },
-      ));
+  let pkg_nvs = packages
+    .iter()
+    .map(|p| p.pkg_id.nv.clone())
+    .collect::<Vec<_>>();
+  let get_version_infos = || {
+    FuturesOrdered::from_iter(pkg_nvs.iter().map(|nv| async move {
+      let package_info = api.package_info(&nv.name).await?;
+      match package_info.version_info(nv) {
+        Ok(version_info) => Ok(version_info),
+        Err(err) => {
+          bail!("Could not find '{}' specified in the lockfile.", err.0);
+        }
+      }
+    }))
+  };
+  let mut version_infos = get_version_infos();
 
   let mut i = 0;
-  while let Some(version_info) = version_infos.next().await {
-    packages[i].dist = version_info?.dist;
+  while let Some(result) = version_infos.next().await {
+    packages[i].dist = match result {
+      Ok(version_info) => version_info.dist,
+      Err(err) => {
+        if api.mark_force_reload() {
+          // reset and try again
+          version_infos = get_version_infos();
+          i = 0;
+          continue;
```

Comment:
Actually looking at it once again - won't this cause infinite loop?

Reply:
Nope. `mark_force_reload` will only return `true` once.

Reply:
Maybe I'm missing what you're seeing though. Could you elaborate?